src/systems/a320_systems/src/pneumatic/wing_anti_ice.rs

```diff
@@ -0,0 +1,1046 @@
+pub struct WingAntiIceValveController {
+    wing_anti_ice_button_pos: WingAntiIcePushButtonMode, //The position of the button
+    valve_pid: Pid<f64>, //PID controller for the valve - to regulate pressure
+    valve_pid_output: f64, //Output of the PID controller - open_amount
+    is_on_ground: bool, //Needed for the 30 seconds test logic
+    system_test_timer: Duration, //Timer to count up to 30 seconds
+    system_test_done: bool, //Timer reached 30 seconds while on the ground
+    controller_signals_on: bool, //If button is pushed and the test is finished
+                                        //the ON light should turn off.
+    supplier_pressurized: bool,
+}
+
+impl WingAntiIceValveController {
+    const WAI_TEST_TIME: Duration = Duration::from_secs(30);
+    pub fn new() -> Self {
+        Self {
+            wing_anti_ice_button_pos: WingAntiIcePushButtonMode::Off,
+            valve_pid: Pid::new(0.5,0.0,0.5,1.,1.,1.,1.,22.),
+            valve_pid_output: 0.,
+            is_on_ground: true,
+            system_test_timer: Duration::from_secs(0),
+            system_test_done: false,
+            controller_signals_on: false,
+            supplier_pressurized: false,
+       }
+    
+    }
+
+    pub fn controller_signals_on(&self) -> bool {
+        self.controller_signals_on
+    }
+    
+    pub fn update (
+        &mut self,
+        context: &UpdateContext,
+        wing_anti_ice_button_pos: WingAntiIcePushButtonMode,
+        supplier_pressurized: bool,
+    ) {
+        self.wing_anti_ice_button_pos = wing_anti_ice_button_pos;
+        self.supplier_pressurized = supplier_pressurized;
+
+        if self.wing_anti_ice_button_pos == WingAntiIcePushButtonMode::On {
+            if self.is_on_ground && self.system_test_done == false {
+                if self.supplier_pressurized {
+                    self.system_test_timer += context.delta();
+                }
+                self.system_test_timer = self.system_test_timer.min(Self::WAI_TEST_TIME);
+                if self.system_test_timer == Self::WAI_TEST_TIME {
+                    self.system_test_done = true;
+                    self.controller_signals_on = false;
+                } else {
+                    self.controller_signals_on = true;
+                }
+            } else if self.is_on_ground == false {
+                self.controller_signals_on = true;
+            }
+        } else {
+            self.controller_signals_on = false;
+        }
+
+        
+        //If the plane has took off, we reset the timer
+        //and set test_done to false in order for the 
+        //mechanism to work when landing.
+        if self.is_on_ground == false && self.system_test_timer > Duration::from_secs(0) {
+            self.system_test_timer = Duration::from_secs(0);
+            self.system_test_done = false;
+        }
+    }
+
+
+    pub fn get_timer(&self) -> Duration {
+        self.system_test_timer
+    }
+}
```

Comment:
I think it would be a good idea to use a basic state machine here to avoid having to nest `if` statements like this. This makes the transitions between states clearer in my eyes. It also avoids any ambiguity with what `system_test_timer` while it is not active.

Note that we cannot have a `pub fn get_timer` anymore as a result of this, so you might have to update some of your unit tests.